File: Server/src/main/content/global/skill/production/runecrafting/TiaraPlugin.java
package content.global.skill.production.runecrafting;

import core.cache.def.impl.ItemDefinition;
import core.cache.def.impl.SceneryDefinition;
import core.game.node.entity.player.Player;
import core.game.node.item.Item;
import core.plugin.Plugin;

import static core.api.ContentAPIKt.setVarbit;
import static core.api.ContentAPIKt.setVarp;

/**
 * The Tiara plugin.
 */
public final class TiaraPlugin implements Plugin<Object> {
    private static final int CONFIG = 491;
    private static final int[] IDS = new int[]{5527, 5529, 5531, 5533, 5535, 5537, 5539, 5541, 5543, 5545, 5547, 5551, 5549};

    @Override
    public Plugin<Object> newInstance(Object arg) throws Throwable {
        for (int i : IDS) {
            ItemDefinition.forId(i).getHandlers().put("equipment", this);
        }
        return this;
    }

    @Override
    public Object fireEvent(String identifier, Object... args) {
        final Player player = (Player) args[0];
        final Item item = (Item) args[1];
        switch (identifier) {
            case "equip":
                MysteriousRuin ruin = MysteriousRuin.forTalisman(Tiara.forItem(item).getTalisman());
                setVarbit(player, SceneryDefinition.forId(ruin.getObject()[0]).getVarbitID(), 1, true);
                break;
            case "unequip":
                final Item other = args.length == 2 ? null : (Item) args[2];
                if (other != null) {
                    Tiara tiara = Tiara.forItem(other);
                    if (tiara != null) {
                        MysteriousRuin r = MysteriousRuin.forTalisman(tiara.getTalisman());
                        setVarbit(player, SceneryDefinition.forId(r.getObject()[0]).getVarbitID(), 1, true);
                        break;
                    }
                }
                setVarp(player, CONFIG, 0);
                break;
        }
        return true;
    }

}

File: Server/src/main/content/global/skill/production/runecrafting/items/tiara/TiaraPlugin.kt
package content.global.skill.production.runecrafting.items.tiara

import content.global.skill.production.runecrafting.data.MysteriousRuin.Companion.forTalisman
import content.global.skill.production.runecrafting.data.Tiara.Companion.forItem
import core.api.setVarbit
import core.api.setVarp
import core.cache.def.impl.ItemDefinition
import core.cache.def.impl.SceneryDefinition
import core.game.node.entity.player.Player
import core.game.node.item.Item
import core.plugin.Plugin

class TiaraPlugin : Plugin<Any?> {

    override fun newInstance(arg: Any?): Plugin<Any?> {
        for (i in IDS) {
            ItemDefinition.forId(i).handlers["equipment"] = this
        }
        return this
    }

    override fun fireEvent(identifier: String, vararg args: Any): Any {
        val player = args[0] as Player
        val item = args[1] as Item
        when (identifier) {
            "equip" -> {
                val ruin = forTalisman(forItem(item)!!.talisman!!)
                setVarbit(player, SceneryDefinition.forId(ruin!!.scenery[0]).varbitID, 1, true)
            }

            "unequip" -> {
                val other = if (args.size == 2) null else args[2] as Item
                if (other != null) {
                    val tiara = forItem(other)
                    if (tiara != null) {
                        val r = forTalisman(tiara.talisman!!)
                        setVarbit(player, SceneryDefinition.forId(r!!.scenery[0]).varbitID, 1, true)
                    }
                }
                setVarp(player, CONFIG, 0)
            }
        }
        return true
    }

    companion object {
        private const val CONFIG = 491
        private val IDS = intArrayOf(5527, 5529, 5531, 5533, 5535, 5537, 5539, 5541, 5543, 5545, 5547, 5549)
    }
}
